apps/total-typescript/src/hooks/use-progress.ts

```diff
@@ -1,90 +1,60 @@
 import React from 'react'
-import {
-  first,
-  filter,
-  find,
-  flatMap,
-  get,
-  indexOf,
-  isEmpty,
-  map,
-  isArray,
-} from 'lodash'
+import {first, filter, find, flatMap, get, indexOf, isEmpty} from 'lodash'
 import {getNextSection} from '@skillrecordings/skill-lesson/utils/get-next-section'
 import {Lesson} from '@skillrecordings/skill-lesson/schemas/lesson'
 import {LessonProgress} from '@skillrecordings/database'
-import {Exercise} from '@skillrecordings/skill-lesson/schemas/exercise'
 import {Section} from '@skillrecordings/skill-lesson/schemas/section'
 import {Module} from '@skillrecordings/skill-lesson/schemas/module'
 import {trpcSkillLessons} from '@skillrecordings/skill-lesson/utils/trpc-skill-lessons'
 
-export function extractExercisesAndResource(sections?: Section[] | null) {
-  const exercises = flatMap<Section, Exercise>(
+export function getAllLessons(sections?: Section[] | null) {
```

Comment:
I tore this hook down a lot and converted any mention of "exercise" to "lesson"

core philosophy here: it's ALL lessons

shouldn't need to determine if it's a solution, explainer, whatever

this means that we need to determine progress based on lesson IDs instead of storing solution slugs, that was broken approach, added loads of complexity